pkg/inframanager/p4/policy.go

```diff
@@ -0,0 +1,866 @@
+				[][]byte{valueToBytes16(portrange[0]),
+					valueToBytes16(portrange[1]),
+					valueToBytes16(portrange[2]),
+					valueToBytes16(portrange[3]),
+					valueToBytes16(portrange[4]),
+					valueToBytes16(portrange[5]),
+					valueToBytes16(portrange[6]),
+					valueToBytes16(portrange[7]),
+					valueToBytes16(portrange[8]),
+					valueToBytes16(portrange[9]),
+					valueToBytes16(portrange[10]),
+					valueToBytes16(portrange[11]),
+					valueToBytes16(portrange[12]),
+					valueToBytes16(portrange[13]),
+					valueToBytes16(portrange[14]),
+					valueToBytes16(portrange[15])}),
+			nil,
+		)
+		if err := p4RtC.InsertTableEntry(ctx, entryAdd); err != nil {
+			log.Errorf("Cannot insert entry into 'udp_dport_rc_table': %v", err)
+			return err
+		}
+
+	case Update:
+		entryMod := p4RtC.NewTableEntry(
+			"k8s_dp_control.udp_dport_rc_table",
+			map[string]client.MatchInterface{
+				"meta.acl_pol_id": &client.ExactMatch{
+					Value: valueToBytes16(polID),
+				},
+			},
+			p4RtC.NewTableActionDirect("k8s_dp_control.do_range_check_udp",
+				[][]byte{valueToBytes16(portrange[0]),
+					valueToBytes16(portrange[1]),
+					valueToBytes16(portrange[2]),
+					valueToBytes16(portrange[3]),
+					valueToBytes16(portrange[4]),
+					valueToBytes16(portrange[5]),
+					valueToBytes16(portrange[6]),
+					valueToBytes16(portrange[7]),
+					valueToBytes16(portrange[8]),
+					valueToBytes16(portrange[9]),
+					valueToBytes16(portrange[10]),
+					valueToBytes16(portrange[11]),
+					valueToBytes16(portrange[12]),
+					valueToBytes16(portrange[13]),
+					valueToBytes16(portrange[14]),
+					valueToBytes16(portrange[15])}),
+			nil,
+		)
+		if err := p4RtC.ModifyTableEntry(ctx, entryMod); err != nil {
+			log.Errorf("Cannot update entry in 'udp_dport_rc_table': %v", err)
+			return err
+		}
+
+	case Delete:
+		entryDelete := p4RtC.NewTableEntry(
+			"k8s_dp_control.udp_dport_rc_table",
+			map[string]client.MatchInterface{
+				"meta.acl_pol_id": &client.ExactMatch{
+					Value: valueToBytes16(polID),
+				},
+			},
+			nil,
+			nil,
+		)
+		if err := p4RtC.DeleteTableEntry(ctx, entryDelete); err != nil {
+			log.Errorf("Cannot delete entry from 'udp_dport_rc_table': %v", err)
+			return err
+		}
+	}
+
+	return nil
+}
+
+func IsNamePresent(substr string, strslice []string) bool {
+	for _, str := range strslice {
+		if strings.Contains(str, substr) {
+			return true
+		}
+	}
+	log.Infof("name %s is not present in given slice", substr)
+	return false
+}
+
+func IsSame(slice1 []uint16, slice2 []uint16) bool {
+	for i := range slice1 {
+		if slice1[i] != slice2[i] {
+			log.Infof("%d and %d are not same", slice1[i], slice2[i])
+			return false
+		}
+	}
+	return true
+}
+
+func InsertPolicyTableEntries(ctx context.Context, p4RtC *client.Client, tbltype tabletype, policy *store.Policy, workloadep *store.PolicyWorkerEndPoint) error {
+	var err error
+	switch tbltype {
+	case policyadd:
+		for ipsetidx, _ := range policy.IpSetIDx {
```

Comment:
The term IpSetIDx is causing confusion as it conflicts with the similar term i.e. IPSet that Calico uses to represent a set of pod IP addresses. Should we call it subPolicyIdx because it is a part of policy (for specific direction and L4 protocol) which can encompass multiple rules within a kubernetes defined or kubernetes level policy.

Reply:
if all agree with this terminology, then i can i change it